pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1189,8 +1189,13 @@ int parser_consume_rows(parser_t *self, size_t nrows) {
 
     /* cannot guarantee that nrows + 1 has been observed */
     word_deletions = self->line_start[nrows - 1] + self->line_fields[nrows - 1];
-    char_count = (self->word_starts[word_deletions - 1] +
-                  strlen(self->words[word_deletions - 1]) + 1);
+    if (word_deletions >= 1) {
```

Comment:
Do you have a handle on why this is required in the first place? I think there is a more general problem with the statement management that this works around (maybe OK)

Reply:
As I have said in the ticket, I am also not certain *why* this works (apart from the obvious fact that this guards against a negative index in the array).
I took the heuristic route and created a test and fiddled around with a couple of options (setting charcount to 0 and to 1).
As it seemed to work and did not break any other unit tests, I thought that it would be useful to submit as a PR.
However, I agree that it does not feel good not knowing exactly *why* this works.
Maybe I put too much trust in unit testing, in the sense that this may break other functionality without unit tests.

Reply:
@roberthdevries : Is there a test that you added which fails when `char_count = 0` ?

Reply:
if `char_count` is 0, there will be an endless loop. So that option was abandoned :smile:

Reply:
With respect to the test you wrote, this loop would manifest for all values of `nrows`

Also, add a comment about `char_count` needing to be at least one.

Reply:
The original ticket only mentions `nrows=1`. I cannot guarantee that the bug would trigger on other values, only the test would not progress further than the first one.
I added the other numbers of `nrows` to be sure that my fix did not introduce unwanted behavior when multiple rows were being read.

I did some more experiments and the value that makes some kind of sense to me is `self->stream_len`. If there are no `word_deletions`, the parser must consume something, and the maximum that can be skipped is the `stream_len`.
With this value all the tests in the `pandas/tests/io/parser` directory keep working.

Reply:
The fact you can make that modification and still have tests passing concerns me a little.  It is hard to tell whether or not there is coverage for the `else` case.

To confirm, if you set `char_count = 0` in the `else` block, do you get test failures?  If not, you may need to add a `sprintf` debug command to see when this case ever gets hit.

Reply:
I tested this case, and it passes the test just fine. (Not sure why it failed in the past).
I am investigating this a bit more, stay tuned. But for now it does not seem to matter much which value I put in there, as long as it is smaller or equal to `stream_len`

Reply:
The patch is changed to `char_count = 0`. As `word_deletions == 0`, and `char_count` is used to skip characters, it seems logical to me that if `word_deletions == 0` that `char_count` must be 0 as well.

Reply:
I have to say that it's still a little weird that we can't pinpoint an exact reason for setting the value of `char_count`, but at least this number I can understand.  Can you add a comment to explain that reasoning?

Reply:
Added